models/channel_helpers.go

```diff
@@ -0,0 +1,258 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package models
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/reform.v1"
+)
+
+// SaveChannel persists notification channel.
+func SaveChannel(q *reform.Querier, c *Channel) error {
+	if err := ValidateChannel(c); err != nil {
+		return err
+	}
+
+	nc, err := channelToNotificationChannel(c)
+	if err != nil {
+		return err
+	}
+
+	err = q.Insert(nc)
+	if err != nil {
+		return errors.Wrap(err, "failed to create notifications channel")
+	}
+
+	return nil
+}
+
+// UpdateChannel updates existing notifications channel.
+func UpdateChannel(q *reform.Querier, c *Channel) error {
+	if err := ValidateChannel(c); err != nil {
+		return errors.Wrap(err, "channel validation failed")
+	}
+
+	nc, err := channelToNotificationChannel(c)
+	if err != nil {
+		return err
+	}
+
+	err = q.Update(nc)
+	if err != nil {
+		return errors.Wrap(err, "failed to create notifications channel")
+	}
+
+	return nil
+}
+
+// RemoveChannel removes notification channel with specified id.
+func RemoveChannel(q *reform.Querier, id string) error {
+	err := q.Delete(&notificationChannel{ID: id})
+	if err != nil {
+		return errors.Wrap(err, "failed to delete notifications channel")
+	}
+	return nil
+}
+
+// GetChannels returns saved notification channels configuration.
+func GetChannels(q *reform.Querier) ([]Channel, error) {
+	structs, err := q.SelectAllFrom(notificationChannelTable, "")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to select notification channels")
+
+	}
+
+	channels := make([]Channel, len(structs))
+	for i, s := range structs {
+		c, err := notificationChannelToChannel(s.(*notificationChannel))
+		if err != nil {
+			return nil, err
+		}
+		channels[i] = *c
+	}
+
+	return channels, nil
+}
+
+// ValidateChannel validates notification channel.
+func ValidateChannel(ch *Channel) error {
```

Comment:
should we make this function public only for tests? looks like we can use `SaveChannel` in those tests

Reply:
Done